src/modules/75-cd/pages/get-started-with-cd/CDOnboardingWizardWithCLI/Steps/VerificationComponents/VerifyGitopsEntities.tsx

```diff
@@ -0,0 +1,199 @@
+    return {
+      icon: 'danger-icon',
+      iconProps: { padding: { right: 'medium' }, color: Color.ERROR, className: css.redIcon }
+    }
+  }, [])
+
+  const loadingIconProps = React.useMemo((): { icon: IconName; iconProps: Omit<IconProps, 'name'> } => {
+    return {
+      icon: 'steps-spinner',
+      iconProps: { padding: { right: 'medium' } }
+    }
+  }, [])
+
+  useEffect(() => {
+    appData?.name &&
+      trackEvent(ONBOARDING_INTERACTIONS.APP_VERIFICATION_SUCCESS, getBranchingProps(stepsProgress, getString))
+    repoData?.repository &&
+      trackEvent(ONBOARDING_INTERACTIONS.REPO_VERIFICATION_SUCCESS, getBranchingProps(stepsProgress, getString))
+    clusterData?.cluster &&
+      trackEvent(ONBOARDING_INTERACTIONS.CLUSTER_VERIFICATION_SUCCESS, getBranchingProps(stepsProgress, getString))
+  }, [appData, clusterData, repoData])
+
+  useEffect(() => {
+    trackEvent(ONBOARDING_INTERACTIONS.APP_VERIFICATION_START, getBranchingProps(stepsProgress, getString))
+    trackEvent(ONBOARDING_INTERACTIONS.REPO_VERIFICATION_START, getBranchingProps(stepsProgress, getString))
+    trackEvent(ONBOARDING_INTERACTIONS.CLUSTER_VERIFICATION_START, getBranchingProps(stepsProgress, getString))
+  }, [])
+
+  useEffect(() => {
+    const allChecksPassed = [appFailed, repoFailed, clusterFailed].every(erorr => erorr === null)
+    const isEntityLoading = [applicationLoading, repoLoading, clusterLoading].includes(true)
+    if (allChecksPassed && !isEntityLoading && !setupState?.gitopsEntitiesVerified) {
+      saveProgress(CDOnboardingSteps.DEPLOYMENT_STEPS, {
+        gitopsEntitiesVerified: true,
+        pipelineVerified: false
+      } as PipelineSetupState)
+    } else if (!allChecksPassed && !isEntityLoading && setupState?.gitopsEntitiesVerified) {
+      saveProgress(CDOnboardingSteps.DEPLOYMENT_STEPS, {
+        gitopsEntitiesVerified: false,
+        pipelineVerified: false
+      } as PipelineSetupState)
+    }
+
+    appFailed && trackEvent(ONBOARDING_INTERACTIONS.APP_VERIFICATION_FAIL, getBranchingProps(stepsProgress, getString))
+    repoFailed &&
+      trackEvent(ONBOARDING_INTERACTIONS.REPO_VERIFICATION_FAIL, getBranchingProps(stepsProgress, getString))
+    clusterFailed &&
+      trackEvent(ONBOARDING_INTERACTIONS.CLUSTER_VERIFICATION_FAIL, getBranchingProps(stepsProgress, getString))
+  }, [
+    appFailed,
+    repoFailed,
+    clusterFailed,
+    applicationLoading,
+    repoLoading,
+    clusterLoading,
+    setupState?.gitopsEntitiesVerified
+  ])
+
+  const retryVerification = (): void => {
+    clusterFailed && refetchCluster()
+    appFailed && refetchApp()
+    repoFailed && refetchRepo()
+  }
+
+  return (
+    <Layout.Vertical spacing={'medium'}>
+      <Text color={Color.BLACK} font={{ variation: FontVariation.FORM_TITLE }}>
+        <String
+          color={Color.BLACK}
+          stringID="cd.getStartedWithCD.flowByQuestions.deploymentSteps.steps.configureGitopsStep.verifyEntities"
+        />
```

Comment:
Any reason to not just use `getString`?
```suggestion
        {getString('cd.getStartedWithCD.flowByQuestions.deploymentSteps.steps.configureGitopsStep.verifyEntities')}
```